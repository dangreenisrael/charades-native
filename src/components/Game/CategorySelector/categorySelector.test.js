import 'react-native';
import {shallow, mount, render} from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import {CategoryButton, mapDispatchToProps} from '.';

describe('CategorySelector', () => {
  const setCategory = jest.fn();
  const Component = shallow(
    <CategoryButton category="Movies" currentCategory="Thing" setCategory={setCategory} />
  );
  Component.simulate('press');

  it('Renders Correctly', () => {
    expect(toJson(Component)).toMatchSnapshot();
  });
  it('Calls setCategory correctly', () => {
    expect(setCategory).toHaveBeenCalledWith('Movies');
  });
  it('Maps dispatch properly', () => {
    const dispatch = jest.fn();
    const props = mapDispatchToProps(dispatch);
    props.setCategory('Foo');
    expect(dispatch).toHaveBeenCalledWith({category: 'Foo', type: 'SET_CATEGORY'});
  });
});
