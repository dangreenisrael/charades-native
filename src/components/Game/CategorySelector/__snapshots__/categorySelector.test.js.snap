// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategorySelector Renders Correctly 1`] = `
<Styled(Button)
  active={false}
  onPress={[Function]}
  title="Movies"
>
  <Styled(Text)>
    Movies
  </Styled(Text)>
  <Styled(Icon)
    name="ios-speedometer-outline"
  />
</Styled(Button)>
`;
